os/kernel/init/os_start.c

```diff
@@ -599,6 +600,11 @@ void os_start(void)
 
 		/* Perform any processor-specific idle state operations */
 
+#ifdef CONFIG_PM
+		pm_idle();
+#endif
+		/* Sleep until an interrupt occurs */
+
 		up_idle();
```

Comment:
You said
```
thapav 7 hours ago  Member
This is provided by the SOC, but up_cpuidle() is called in case power framework is not supported.
```
But after calling pm_idle, up_idle will be called.

Reply:
Yes , based on review comment by @manoharahk , I am calling it always and not only in else case